packages/api/lambdas/internal-reconciliation-report.js

```diff
@@ -193,8 +217,28 @@ async function getCollectionsForGranules(recReportParams) {
   const {
     granuleIds,
   } = recReportParams;
-
-  const dbCollectionIds = await getAllCollectionIdsByGranuleIds(recReportParams);
+  let dbCollectionIds = [];
+
+  await pRetry(
+    async () => {
+      try {
+        dbCollectionIds = await getAllCollectionIdsByGranuleIds(recReportParams);
```

Comment:
retry at the lowest level function when possible.

Reply:
Do we know which reconciliation report function/operation throws `Connection terminated unexpectedly`, so we can avoid adding `await pRetry` everywhere?

Reply:
For the user, it occurred in the `Inventory` report. I don't think this issue is limited to the report type, though. It's a general knex issue so it can happen when generating any report. @jennyhliu